great_expectations/datasource/fluent/data_asset/data_connector/azure_blob_storage_data_connector.py

```diff
@@ -27,6 +27,7 @@ class _AzureOptions(pydantic.BaseModel):
     abs_container: str
     abs_name_starts_with: str = ""
     abs_delimiter: str = "/"
+    abs_recursive_file_discovery: bool = False
```

Comment:
@toivomattila -- when convenient, could you please also add this new flag to the docstrings (in every `DataConnector` and other relevant modules).  Thank you very much.